### ISSUE TITLE
9.0.0-rc.5: angular/compiler and angular/compiler-cli throw errors when running ngcc

### ISSUE THREAD

**Issue by mikolaj-leszczynski** [angular/angular#34275]
### Affected Package
**@angular/compiler
@angular/compiler-cli**

### Is this a regression?

Yes

### Description

When @angular/compiler is on `9.0.0-rc.5`  when running `yarn ngcc` I'm getting the compile error. There is also another compile error when @angular/compiler-cli is on `9.0.0-rc.5`

## 🔥 Exception or Error
When @angular/compiler is on `9.0.0-rc.5` 
<pre><code>
Compiling @angular/cdk/keycodes : es2015 as esm2015
Compiling @angular/animations : fesm5 as esm5
Compiling @angular/animations : fesm2015 as esm2015
Compiling @angular/animations : esm5 as esm5
Compiling @angular/animations : esm2015 as esm2015
Compiling @angular/animations : main as umd
Compiling @angular/compiler/testing : fesm2015 as esm2015
Compiling @angular/compiler/testing : fesm2015 as esm2015
Compiling apollo-angular-link-http-common : fesm2015 as esm2015
Compiling @angular/cdk/testing : es2015 as esm2015
Compiling @angular/cdk/testing : main as umd
Compiling @angular/cdk/testing : module as esm5
Compiling @angular/cdk/keycodes : main as umd
Compiling @angular/cdk/keycodes : module as esm5
Compiling @angular/compiler/testing : fesm5 as esm5
Compiling @angular/core : fesm5 as esm5
Compiling @angular/core : fesm2015 as esm2015
Compiling @angular/core : fesm2015 as esm2015
Compiling @angular/core : main as umd
Compiling @angular/compiler/testing : esm2015 as esm2015
Compiling @angular/core : esm2015 as esm2015
Compiling @angular/core : esm5 as esm5
Compiling @angular/compiler/testing : main as umd
Compiling @angular/compiler/testing : esm5 as esm5
Compiling @angular/animations/browser : fesm2015 as esm2015
Compiling @angular/cdk : es2015 as esm2015
Compiling @angular/cdk/coercion : es2015 as esm2015
Compiling @angular/common : fesm2015 as esm2015
Compiling @angular/common : fesm2015 as esm2015
Compiling @angular/core : fesm5 as esm5
Compiling @angular/core/testing : fesm2015 as esm2015
Error: Error on worker #1: Error: Import unknown module or symbol [object Object]
    at ExpressionTranslatorVisitor.visitExternalExpr (/path/to/my/repo/node_modules/@angular/compiler-cli/src/ngtsc/translator/src/translator.js:211:23)
    at ExternalExpr.visitExpression (/path/to/my/repo/node_modules/@angular/compiler/bundles/compiler.umd.js:1663:28)
    at ExpressionTranslatorVisitor.visitTypeofExpr (/path/to/my/repo/node_modules/@angular/compiler-cli/src/ngtsc/translator/src/translator.js:272:45)
    at TypeofExpr.visitExpression (/path/to/my/repo/node_modules/@angular/compiler/bundles/compiler.umd.js:1390:28)
    at ExpressionTranslatorVisitor.visitBinaryOperatorExpr (/path/to/my/repo/node_modules/@angular/compiler-cli/src/ngtsc/translator/src/translator.js:241:44)
    at BinaryOperatorExpr.visitExpression (/path/to/my/repo/node_modules/@angular/compiler/bundles/compiler.umd.js:1792:28)
    at ExpressionTranslatorVisitor.visitBinaryOperatorExpr (/path/to/my/repo/node_modules/@angular/compiler-cli/src/ngtsc/translator/src/translator.js:241:44)
    at BinaryOperatorExpr.visitExpression (/path/to/my/repo/node_modules/@angular/compiler/bundles/compiler.umd.js:1792:28)
    at ExpressionTranslatorVisitor.visitBinaryOperatorExpr (/path/to/my/repo/node_modules/@angular/compiler-cli/src/ngtsc/translator/src/translator.js:241:44)
    at BinaryOperatorExpr.visitExpression (/path/to/my/repo/node_modules/@angular/compiler/bundles/compiler.umd.js:1792:28)
    at ClusterMaster.onWorkerMessage (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:158:27)
    at /path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:46:95
    at ClusterMaster.<anonymous> (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:238:57)
    at step (/path/to/my/repo/node_modules/tslib/tslib.js:136:27)
    at Object.next (/path/to/my/repo/node_modules/tslib/tslib.js:117:57)
    at /path/to/my/repo/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/path/to/my/repo/node_modules/tslib/tslib.js:106:16)
    at EventEmitter.<anonymous> (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:232:32)
    at EventEmitter.emit (events.js:198:13)
Compiling @angular/cdk/collections : es2015 as esm2015
Compiling @angular/cdk/portal : es2015 as esm2015
Compiling @angular/cdk/observers : es2015 as esm2015
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre></code>

When  @angular/compiler-cli is on `9.0.0-rc.5` :
<pre><code>
Compiling @angular/cdk/keycodes : es2015 as esm2015
Compiling @angular/animations : fesm5 as esm5
Compiling @angular/animations : fesm2015 as esm2015
Compiling @angular/animations : esm2015 as esm2015
Compiling @angular/animations : esm5 as esm5
Compiling @angular/animations : main as umd
Compiling @angular/compiler/testing : fesm2015 as esm2015
Compiling @angular/compiler/testing : fesm2015 as esm2015
Compiling apollo-angular-link-http-common : fesm2015 as esm2015
Compiling @angular/cdk/testing : es2015 as esm2015
Compiling @angular/cdk/testing : main as umd
Compiling @angular/cdk/testing : module as esm5
Compiling @angular/cdk/keycodes : main as umd
Compiling @angular/cdk/keycodes : module as esm5
Compiling @angular/compiler/testing : fesm5 as esm5
Compiling @angular/core : fesm5 as esm5
Compiling @angular/core : fesm2015 as esm2015
Compiling @angular/core : fesm2015 as esm2015
Compiling @angular/compiler/testing : esm2015 as esm2015
Compiling @angular/core : main as umd
Compiling @angular/compiler/testing : main as umd
Compiling @angular/core : esm2015 as esm2015
Compiling @angular/core : esm5 as esm5
Compiling @angular/compiler/testing : esm5 as esm5
Compiling @angular/cdk/coercion : es2015 as esm2015
Compiling @angular/cdk/collections : es2015 as esm2015
Compiling @angular/cdk : es2015 as esm2015
Compiling @angular/animations/browser : fesm2015 as esm2015
Compiling @angular/core : fesm5 as esm5
Compiling @angular/common : fesm2015 as esm2015
Compiling @angular/cdk/portal : es2015 as esm2015
Compiling @angular/cdk/observers : es2015 as esm2015
Compiling @angular/common : fesm2015 as esm2015
Compiling @ngrx/store : fesm2015 as esm2015
Compiling @angular/platform-browser : fesm2015 as esm2015
Compiling @angular/cdk/platform : es2015 as esm2015
Compiling @angular/core : esm2015 as esm2015
Compiling @angular/cdk/bidi : es2015 as esm2015
Compiling @angular/forms : fesm2015 as esm2015
Compiling @angular/platform-browser/animations : fesm2015 as esm2015
Compiling @angular/common/http : fesm2015 as esm2015
Compiling @angular/cdk/a11y : es2015 as esm2015
Compiling @angular/platform-browser : fesm2015 as esm2015
Compiling @angular/router : fesm2015 as esm2015
Compiling @angular/cdk/scrolling : es2015 as esm2015
Compiling @angular/upgrade/static : fesm2015 as esm2015
Compiling @angular/cdk/layout : es2015 as esm2015
Compiling @angular/core/testing : fesm2015 as esm2015
Compiling @angular/material/core : es2015 as esm2015
Compiling @angular/cdk/accordion : es2015 as esm2015
Compiling @angular/cdk/text-field : es2015 as esm2015
Compiling @angular/cdk/overlay : es2015 as esm2015
Compiling @angular/common/http : fesm2015 as esm2015
Compiling @angular/platform-browser-dynamic : fesm2015 as esm2015
Compiling @angular/material/divider : es2015 as esm2015
Compiling @angular/cdk/stepper : es2015 as esm2015
Compiling @angular/cdk/table : es2015 as esm2015
Compiling @angular/cdk/tree : es2015 as esm2015
Compiling @angular/material/button : es2015 as esm2015
Compiling @angular/material/form-field : es2015 as esm2015
Compiling @angular/material/dialog : es2015 as esm2015
Compiling @angular/platform-browser/testing : fesm2015 as esm2015
Compiling @angular/common/testing : fesm2015 as esm2015
Compiling @angular/material/icon : es2015 as esm2015
Compiling @angular/material/tooltip : es2015 as esm2015
Compiling @angular/material/input : es2015 as esm2015
Compiling @angular/core/testing : fesm2015 as esm2015
Compiling @angular/platform-browser/testing : fesm2015 as esm2015
Compiling @angular/material/select : es2015 as esm2015
Compiling @ngrx/effects : fesm2015 as esm2015
Compiling @angular/platform-browser-dynamic : fesm2015 as esm2015
Compiling @angular/material/badge : es2015 as esm2015
Compiling @angular/material/card : es2015 as esm2015
Compiling @angular/material/autocomplete : es2015 as esm2015
Compiling @angular/material/bottom-sheet : es2015 as esm2015
Compiling @angular/material/checkbox : es2015 as esm2015
Compiling apollo-angular : fesm2015 as esm2015
Compiling @angular/material/button-toggle : es2015 as esm2015
Compiling @angular/material/chips : es2015 as esm2015
Compiling @angular/material/expansion : es2015 as esm2015
Compiling @angular/material/grid-list : es2015 as esm2015
Compiling @angular/material/paginator : es2015 as esm2015
Compiling @angular/material/list : es2015 as esm2015
Compiling @angular/material/menu : es2015 as esm2015
Compiling @angular/material/datepicker : es2015 as esm2015
Compiling @angular/material/progress-bar : es2015 as esm2015
Compiling @angular/material/progress-spinner : es2015 as esm2015
Compiling @angular/material/radio : es2015 as esm2015
Compiling @angular/material/sidenav : es2015 as esm2015
Compiling @angular/material/slide-toggle : es2015 as esm2015
Compiling @angular/material/sort : es2015 as esm2015
Compiling @angular/material/slider : es2015 as esm2015
Compiling @angular/material/snack-bar : es2015 as esm2015
Compiling @angular/material/stepper : es2015 as esm2015
Compiling @angular/material/table : es2015 as esm2015
Compiling @angular/animations/browser : fesm5 as esm5
Compiling @angular/material/toolbar : es2015 as esm2015
Compiling @angular/animations/browser/testing : fesm2015 as esm2015
Compiling @angular/animations/browser/testing : fesm5 as esm5
Compiling @angular/animations/browser : esm2015 as esm2015
Compiling @angular/animations/browser/testing : esm2015 as esm2015
Compiling @angular/material/tabs : es2015 as esm2015
Compiling @angular/animations/browser/testing : esm5 as esm5
Compiling @angular/animations/browser/testing : main as umd
Compiling @angular/animations/browser : main as umd
Compiling @angular/material/tree : es2015 as esm2015
Compiling @angular/animations/browser : esm5 as esm5
Compiling @angular/common : fesm5 as esm5
Compiling @angular/common : esm5 as esm5
Compiling @angular/common : esm2015 as esm2015
Compiling @angular/cdk/platform : module as esm5
Compiling @angular/cdk/coercion : main as umd
Compiling @angular/cdk/platform : main as umd
Compiling @angular/common : main as umd
Compiling @angular/cdk/bidi : module as esm5
Compiling @angular/cdk/coercion : module as esm5
Compiling @angular/cdk/bidi : main as umd
Compiling @angular/cdk/collections : main as umd
Compiling @angular/cdk/collections : module as esm5
Compiling @angular/cdk/scrolling : module as esm5
Compiling @angular/cdk/scrolling : main as umd
Compiling @angular/common/http : fesm5 as esm5
Compiling @angular/cdk/drag-drop : module as esm5
Compiling @angular/common/http : esm2015 as esm2015
Compiling @angular/cdk/drag-drop : main as umd
Compiling @angular/cdk/drag-drop : es2015 as esm2015
Compiling @angular/common/http/testing : fesm5 as esm5
Compiling @angular/common/http : esm5 as esm5
Compiling @angular/common/http : main as umd
Compiling @angular/common/http/testing : esm2015 as esm2015
Error: Error on worker #7: Error: Failed to compile entry-point @angular/common/http/testing due to compilation errors:
node_modules/@angular/common/http/http.d.ts(2801,22): error TS-996002: Appears in the NgModule.imports of HttpClientTestingModule, but could not be resolved to an NgModule class

    at ClusterWorker.compile (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/main.js:170:27)
    at Worker.<anonymous> (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/worker.js:41:42)
    at Worker.emit (events.js:198:13)
    at process.Worker.process.on (internal/cluster/worker.js:28:12)
    at process.emit (events.js:198:13)
    at emit (internal/child_process.js:832:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at ClusterMaster.onWorkerMessage (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:158:27)
    at /path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:46:95
    at ClusterMaster.<anonymous> (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:238:57)
    at step (/path/to/my/repo/node_modules/tslib/tslib.js:136:27)
    at Object.next (/path/to/my/repo/node_modules/tslib/tslib.js:117:57)
    at /path/to/my/repo/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/path/to/my/repo/node_modules/tslib/tslib.js:106:16)
    at EventEmitter.<anonymous> (/path/to/my/repo/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:232:32)
    at EventEmitter.emit (events.js:198:13)
Compiling @angular/common/http/testing : esm5 as esm5
Compiling @angular/common/http/testing : main as umd
Compiling @angular/platform-browser : fesm5 as esm5
Compiling @angular/common/http/testing : fesm2015 as esm2015
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

</pre></code> 

## 🌍  Your Environment

**Angular Version:**
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.0-rc.5
Node: 10.16.0
OS: linux x64

Angular: 9.0.0-rc.5
... animations, bazel, cli, common, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, upgrade
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.5
@angular-devkit/build-angular     0.900.0-rc.5
@angular-devkit/build-optimizer   0.900.0-rc.5
@angular-devkit/build-webpack     0.900.0-rc.5
@angular-devkit/core              9.0.0-rc.5
@angular-devkit/schematics        9.0.0-rc.5
@angular/cdk                      8.2.3
@angular/compiler                 9.0.0-rc.2
@angular/compiler-cli             9.0.0-rc.2
@angular/material                 8.2.3
@bazel/bazel                      2.0.0-rc3
@bazel/bazel-linux_x64            2.0.0-rc3
@bazel/benchmark-runner           0.1.0
@bazel/buildifier                 0.29.0
@bazel/buildifier-linux_x64       0.29.0
@bazel/hide-bazel-files           0.36.2
@bazel/ibazel                     v0.10.3
@bazel/karma                      0.42.2
@bazel/rollup                     0.42.2
@bazel/terser                     0.42.2
@bazel/typescript                 0.42.2
@ngtools/webpack                  9.0.0-rc.5
@schematics/angular               9.0.0-rc.5
@schematics/update                0.900.0-rc.5
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Comment by EmaGht** [comment]
Corollary question:
Why does the package for rc.5 exists but the latest Changelog.MD doesn't mention it?

**Comment by kKen94** [comment]
Same issue

**Comment by petebacondarwin** [comment]
I tried the following and could not reproduce the error.

```
cd ~
yarn ng global add @angular/cli@next
ng new --defaults=true rc5-test
cd rc5-test
yarn ngcc
```

It completed as expected...

Please can you check that all your Angular versions are in sync. I notice in your `ng version` output you have 

```
@angular/compiler                 9.0.0-rc.2
@angular/compiler-cli             9.0.0-rc.2
```

Then can you provide full steps to reproduce this problem?

Thanks!

**Comment by kKen94** [comment]
Today I simply updated @angular/cdk and @angular/material from 9.0.0-rc.4 to 9.0.0-rc.5.
Also @angular/compiler and @angular/compiler-cli are updated to 9.0.0-rc.5

**Comment by petebacondarwin** [comment]
Can you provide steps for me to reproduce the problem? It is difficult to help otherwise

**Comment by mikolaj-leszczynski** [comment]
@petebacondarwin I'm workin on it :)

**Comment by petebacondarwin** [comment]
Great! Thanks

**Comment by s992** [comment]
@petebacondarwin I'm working with @mikolaj-leszczynski on this and while we still don't have steps to reproduce, I was able to complete ngcc by setting the async flag to false. Not sure if that raises any flags for you, but wanted to drop the info in here.

**Comment by s992** [comment]
@petebacondarwin I'm still having trouble finding a minimal repro and given that it succeeds with parallel compilation disabled I'm thinking it may not be related to any one package.

**Comment by petebacondarwin** [comment]
Can you try using the Angular framework artifacts from this PR : https://github.com/angular/angular/pull/34369

You can get the artifacts here: https://app.circleci.com/jobs/github/angular/angular/562400/artifacts

And see this for more information about how to use them: https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#getting-packages-from-build-artifacts

**Comment by s992** [comment]
@petebacondarwin That link 404s for me, looks like the correct link for that PR is https://circleci.com/gh/angular/angular/562563#artifacts?

Tried it with the artifacts from that link and seeing the same style of error, but a different file:

```
Error: Error on worker #5: Error: Failed to compile entry-point @angular/platform-browser due to compilation errors:
node_modules/@angular/core/core.d.ts:259:22 - error NG6003: Appears in the NgModule.exports of BrowserModule, but could not be resolved to an NgModule, Component, Directive, or Pipe class

259 export declare class ApplicationModule {
                         ~~~~~~~~~~~~~~~~~

    at ClusterWorker.compile (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/main.js:171:27)
    at Worker.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/worker.js:41:42)
    at Worker.emit (events.js:198:13)
    at process.Worker.process.on (internal/cluster/worker.js:28:12)
    at process.emit (events.js:198:13)
    at emit (internal/child_process.js:832:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at ClusterMaster.onWorkerMessage (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:158:27)
    at /Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:46:95
    at ClusterMaster.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:238:57)
    at step (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:136:27)
    at Object.next (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:117:57)
    at /Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:106:16)
    at EventEmitter.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:232:32)
    at EventEmitter.emit (events.js:198:13)
```

Can confirm that running `yarn ngcc --async false` succeeds. Here's our package.json with the new artifacts linked in:

```json
{
  "dependencies": {
    "@angular-devkit/build-angular": "^0.900.0-rc.5",
    "@angular-tracing/zipkin": "^0.0.12",
    "@angular/animations": "https://562563-24195339-gh.circle-artifacts.com/0/angular/animations-pr34369-4d0647d45e.tgz",
    "@angular/cdk": "8.2.3",
    "@angular/common": "https://562563-24195339-gh.circle-artifacts.com/0/angular/common-pr34369-4d0647d45e.tgz",
    "@angular/core": "https://562563-24195339-gh.circle-artifacts.com/0/angular/core-pr34369-4d0647d45e.tgz",
    "@angular/forms": "https://562563-24195339-gh.circle-artifacts.com/0/angular/forms-pr34369-4d0647d45e.tgz",
    "@angular/material": "8.2.3",
    "@angular/platform-browser": "https://562563-24195339-gh.circle-artifacts.com/0/angular/platform-browser-pr34369-4d0647d45e.tgz",
    "@angular/router": "https://562563-24195339-gh.circle-artifacts.com/0/angular/router-pr34369-4d0647d45e.tgz",
    "@angular/upgrade": "https://562563-24195339-gh.circle-artifacts.com/0/angular/upgrade-pr34369-4d0647d45e.tgz",
    "@avatsaev/angular-jwt": "^4.0.2",
    "@ngrx/effects": "8.4.0",
    "@ngrx/router-store": "8.4.0",
    "@ngrx/store": "8.4.0",
    "apollo-angular": "^1.5.0",
    "apollo-angular-link-http": "^1.4.0",
    "apollo-cache-inmemory": "^1.3.9",
    "apollo-client": "^2.4.5",
    "apollo-link": "^1.2.3",
    "browser-process-hrtime": "^1.0.0",
    "bson-objectid": "^1.2.4",
    "core-js": "2.6.9",
    "empty-module": "^0.0.2",
    "enhanced-resolve": "3.3.0",
    "graphql": "^14.0.2",
    "graphql-tag": "^2.10.0",
    "humps": "2.0.0",
    "karma-cli": "^1.0.1",
    "lodash": "^4.17.13",
    "ngrx-store-logger": "^0.1.7",
    "node-sass": "^4.12.0",
    "noop-logger": "^0.1.1",
    "reselect": "^3.0.1",
    "rxjs": "6.5.3",
    "tslib": "1.10.0",
    "zipkin": "^0.14.3",
    "zipkin-transport-http": "^0.14.3",
    "zone.js": "0.10.2"
  },
  "devDependencies": {
    "@angular/bazel": "https://562563-24195339-gh.circle-artifacts.com/0/angular/bazel-pr34369-4d0647d45e.tgz",
    "@angular/cli": "9.0.0-rc.5",
    "@angular/compiler": "https://562563-24195339-gh.circle-artifacts.com/0/angular/compiler-pr34369-4d0647d45e.tgz",
    "@angular/compiler-cli": "https://562563-24195339-gh.circle-artifacts.com/0/angular/compiler-cli-pr34369-4d0647d45e.tgz",
    "@angular/language-service": "https://562563-24195339-gh.circle-artifacts.com/0/angular/language-service-pr34369-4d0647d45e.tgz",
    "@angular/platform-browser-dynamic": "https://562563-24195339-gh.circle-artifacts.com/0/angular/platform-browser-dynamic-pr34369-4d0647d45e.tgz",
    "@babel/cli": "^7.6.0",
    "@babel/core": "^7.6.0",
    "@babel/preset-env": "^7.6.0",
    "@bazel/bazel": "2.0.0-rc3",
    "@bazel/benchmark-runner": "0.1.0",
    "@bazel/buildifier": "^0.29.0",
    "@bazel/hide-bazel-files": "latest",
    "@bazel/ibazel": "^0.10.3",
    "@bazel/karma": "^0.42.2",
    "@bazel/rollup": "^0.42.2",
    "@bazel/terser": "^0.42.2",
    "@bazel/typescript": "^0.42.2",
    "@types/gapi": "0.0.31",
    "@types/gapi.auth2": "0.0.35",
    "@types/jasmine": "2.8.16",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "10.14.22",
    "codelyzer": "^4.2.0",
    "concurrently": "3.5.1",
    "dd-strict-tsc": "^1.0.3",
    "ddts": "^0.5.1",
    "graphql-code-generator": "^0.14.5",
    "graphql-codegen-typescript-apollo-angular": "^0.14.5",
    "graphql-codegen-typescript-client": "^0.14.5",
    "graphql-codegen-typescript-common": "^0.14.5",
    "graphql-codegen-typescript-server": "^0.14.5",
    "husky": "0.14.3",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~1.7.1",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.0",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-summary-reporter": "^1.5.1",
    "lint-staged": "^7.2.0",
    "prettier": "^1.13.7",
    "protractor": "5.4.1",
    "rollup": "^1.21.4",
    "rollup-plugin-commonjs": "^10.1.0",
    "rollup-plugin-node-resolve": "^5.2.0",
    "terser": "4.3.1",
    "ts-node": "~2.0.0",
    "tslint": "^5.10.0",
    "typescript": "3.6.4"
  }
}
```

Using rc.6 throws the same exception while rc.5 is the exception that @mikolaj-leszczynski posted in his original report.

**Comment by s992** [comment]
I think I found the right link (https://circleci.com/gh/angular/angular/562400#artifacts) but still erroring:

```
Error: Error on worker #5: Error: Failed to compile entry-point @angular/common/http/testing (esm2015 as esm2015) due to compilation errors:
node_modules/@angular/common/http/http.d.ts:2801:22 - error NG6002: Appears in the NgModule.imports of HttpClientTestingModule, but could not be resolved to an NgModule class

2801 export declare class HttpClientModule {
                          ~~~~~~~~~~~~~~~~

    at ClusterWorker.compile (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/main.js:171:27)
    at Worker.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/worker.js:41:42)
    at Worker.emit (events.js:198:13)
    at process.Worker.process.on (internal/cluster/worker.js:28:12)
    at process.emit (events.js:198:13)
    at emit (internal/child_process.js:832:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at ClusterMaster.onWorkerMessage (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:158:27)
    at /Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:46:95
    at ClusterMaster.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:238:57)
    at step (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:136:27)
    at Object.next (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:117:57)
    at /Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/seanwalsh/code/admin-dashboard/node_modules/tslib/tslib.js:106:16)
    at EventEmitter.<anonymous> (/Users/seanwalsh/code/admin-dashboard/node_modules/@angular/compiler-cli/ngcc/src/execution/cluster/master.js:232:32)
    at EventEmitter.emit (events.js:198:13)
```

**Comment by kKen94** [comment]
I have the same errors above with postinstall: "ngcc --properties es2015 browser module main --first-only --create-ivy-entry-points". Totally random, there is no logic.
 I usual run the command until it works

**Comment by gkalpak** [comment]
This is likely related to https://github.com/angular/angular/issues/34411#issuecomment-565765622.
Does running `ngcc` with the `--no-async` flag (in addition to other flags you might be using) fix the problem?

**Comment by marcus-sa** [comment]
@gkalpak running without parallelization seems to fix the issues for me as well.

**Comment by gkalpak** [comment]
Thx for verifying, @marcus-sa 👍 

I am going to close this issue (since it seems to have the same root cause as #34411). Please, track #34411 and if the problem persists once the fix for #34411 has been released open a new issue with the problem details.

##
> IMO `async` flag should be off by default since it doesn't work properly and is definitely experimental.

IMO, the fact that there is bug in it does not mean that the async mode should be turned off, nor that it is experimental :grin:
There is a specific bug in ngcc (as described in https://github.com/angular/angular/issues/34411#issuecomment-565765622 - i.e. that ngcc does not detect typings-only dependencies), which happens to affect the parallel (async) mode with a higher probability, but could theoretically affect serial (sync) mode (if I understand correctly). (It is just that it is much more likely to run into the issue in parallel mode.)

Anyway, let's fix the bug and see what else comes up.
